### PR TITLE
Feature/fp 19

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -71,15 +71,11 @@ android {
...
 
             minifyEnabled true
-            useProguard true
 
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
         debug {
-            signingConfig signingConfigs.release
-
             minifyEnabled true
```
Comment (PiotrKlis) [inline example/android/app/build.gradle:78]: As a note, for `debug` there's no need to set m`inifyEnabled` to true and add proguard.

File: example/lib/screens/pin_request_screen.dart [hunk 2]
```diff
@@ -251,11 +249,8 @@ class NumPad extends StatelessWidget {
     return Expanded(
       child: Container(
         margin: EdgeInsets.all(2),
-        child: FlatButton(
-          height: 50,
```
Comment (PiotrKlis) [inline example/lib/screens/pin_request_screen.dart:255]: Not part of this PR but these magic numbers hurt my eyes.

File: ios/Classes/PluginExtensions/SwiftOneginiPlugin+Register.swift [hunk 3]
```diff
@@ -46,8 +46,8 @@ extension SwiftOneginiPlugin: OneginiPluginRegisterProtocol {
         if let _typeValue = typeValue, let value = WebSignInType.init(rawValue: _typeValue) {
             type = value
         }
-        
         OneginiModuleSwift.sharedInstance.handleRegisteredProcessUrl(url ?? "", webSignInType: type)
+        result(nil)
```
Comment (PiotrKlis) [inline ios/Classes/PluginExtensions/SwiftOneginiPlugin+Register.swift:50]: why?

File: lib/onegini.dart [hunk 4]
```diff
@@ -57,8 +57,11 @@ class Onegini {
       });
       eventListener.listen();
       return removedUserProfileListFromJson(removedUserProfiles);
-    } on PlatformException catch (error) {
-      throw error;
+    } on TypeError catch (error) {
```
Comment (PiotrKlis) [inline lib/onegini.dart:60]: Won't that catch **only** `TypeError`? What about handling different errors that can occur? I think we could use dart `on` keyword, like in this example:

```
try {
  // ...
} on SomeException catch(e) {
 //Handle exception of type SomeException
} catch(e) {
 //Handle all other exceptions
}
```

**Review by PiotrKlis — CHANGES_REQUESTED** [review]
I think we're missing bunch of exceptions that can occur and catch only one - TypeError. Other than that it looks ok to me.

File: ios/Classes/PluginExtensions/SwiftOneginiPlugin+Register.swift (hunk 3 above)
Comment (MichielVrins) [reply ios/Classes/PluginExtensions/SwiftOneginiPlugin+Register.swift:50]: This is needed to resovle the promise, without this, the function will just return and never actually resolve leaving the future waiting on the Dart side.

File: example/lib/screens/pin_request_screen.dart (hunk 2 above)
Comment (MichielVrins) [reply example/lib/screens/pin_request_screen.dart:255]: Funny enough, this height is exactly the same as not having it in the replacement TextButton.

File: lib/onegini.dart (hunk 4 above)
Comment (MichielVrins) [reply lib/onegini.dart:60]: As discussed, we will let all other errors bubble up as they will just reject the future, the develop can then handle these as they want.

**Review by PiotrKlis — APPROVED** [review]
